pkg/services/accesscontrol/cacheutils_test.go

```diff
@@ -122,24 +117,31 @@ func TestGetSearchPermissionCacheKey(t *testing.T) {
 				Action: "datasources:read",
 				Scope:  "datasources:*",
 			},
-			expected: "rbac-permissions-1-user-1-datasources:read-datasources:*",
 		},
 		{
-			name: "should return correct key for user with role prefixes",
 			signedInUser: &user.SignedInUser{
 				OrgID:  1,
 				UserID: 1,
 			},
 			searchOptions: SearchOptions{
 				RolePrefixes: []string{"foo", "bar"},
 			},
-			expected: "rbac-permissions-1-user-1-foo-bar",
 		},
 	}
 
-	for _, tc := range testcases {
-		t.Run(tc.name, func(t *testing.T) {
-			assert.Equal(t, tc.expected, GetSearchPermissionCacheKey(tc.signedInUser, tc.searchOptions))
-		})
+	cacheKeys := make([]string, 0, len(keyInputs))
+
+	for _, i := range keyInputs {
+		key, err := GetSearchPermissionCacheKey(testLogger, i.signedInUser, i.searchOptions)
+		require.NoError(t, err)
+		cacheKeys = append(cacheKeys, key)
+	}
+
+	uniqueCheck := make(map[string]bool)
+	for _, str := range cacheKeys {
+		require.False(t, uniqueCheck[str], "Found duplicate string: %s", str)
```

Comment:
Would be nice to add a test case where we expect duplicate keys, just to show that the cache key will allow to retrieve the right entry.

Reply:
Definitely, added the test: 181058c40a94ce002502452e5c021b3cca8ffaa9